staging/src/k8s.io/kubectl/pkg/config/v1alpha1/types.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Preferences stores elements of KubeRC configuration file
+type Preferences struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Overrides is used to change the defaults values of flags of commands.
+	// This is especially useful, when user doesn't want to explicitly
+	// set flags each time.
+	// +listType=atomic
+	Overrides []CommandOverride `json:"overrides"`
+
+	// Aliases stores the alias definitions. If the alias name collides with
+	// a built-in command, built-in command always overrides the alias name.
+	// +listType=atomic
+	Aliases []AliasOverride `json:"aliases"`
+}
+
+// AliasOverride stores the alias definitions.
+// It is applied in a pre-defined order which is
+// kubectl [ALIAS NAME] expands to kubectl [COMMAND] [ARGUMENTS]
```

Comment:
Kuberc flags have no representation in this example, because kuberc flags modify the **default** of flag rather than setting a real value. I've updated the possible ordering in https://github.com/kubernetes/kubernetes/pull/125230/files#diff-6b83ea83ef1e9eac7db92f8078c9453496f21404b162f3300451e2750afac9ceR41 (USER_FLAGS are arbitrary though)